Unit3FinalAssessment/Unit3FinalAssessment/DetailViewController.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  DetailViewController.swift
+//  Unit3FinalAssessment
+//
+//  Created by C4Q on 11/10/16.
+//  Copyright © 2016 C4Q. All rights reserved.
+//
+
+import UIKit
+
+class DetailViewController: UIViewController {
+    
+    var object: MuseumObject?
+    
+
+    
+    @IBOutlet weak var objectImage: UIImageView!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.navigationItem.title = object?.title
+        self.objectImage.image = #imageLiteral(resourceName: "default-placeholder")
+        self.descriptionLabel.text = object?.description
+        
+        guard let validObject = object else {return}
+        
+        APIManager.manager.getData(endPoint: validObject.image) { (data: Data?) in
+            if let d = data {
+                print(d)
+                self.objectImage.image = UIImage(data: d)
+            }
+            DispatchQueue.main.async {
```

Comment:
The only thing you need to update the image is do do the setting on the main queue. You'd done it a line above. I recognize reloadInputViews from other projects where cells weren't updating in a table. The correct call to make there  is cell.view.setNeedsLayout(). This is what I went with for those back to back data calls:

```swift
        APIManager.manager.getData(endPoint: validObject.image) { (data: Data?) in
            DispatchQueue.main.async {
                if let d = data {
                    self.objectImage.image = UIImage(data: d)
                }
            }
        }
        
        APIManager.manager.getData(endPoint: validObject.description) { (data: Data?) in
            if let d = data {
                if let des = Description.getDescription(data: d) {
                    DispatchQueue.main.async {
                        self.descriptionLabel.text = des.decription
                    }
                }
            }
        }
```
Nice work on this, btw.